From 2d6ae243d0eceec718de0ca9a2d6f1453b085de6 Mon Sep 17 00:00:00 2001
From: Yusuke Sato <yusukes@chromium.org>
Date: Fri, 17 Sep 2010 22:26:50 +0900
Subject: [PATCH 07/10] Fix g_assertion failure in bus_ibus_impl_{load,save}_global_engine_name functions.

Add NULL checks to the functions so they don't abort with assertion failure even if the functions are called before the ibus configuration daemon gets ready.
See http://code.google.com/p/chromium-os/issues/detail?id=6689#c27 for details.
---
 bus/ibusimpl.c |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/bus/ibusimpl.c b/bus/ibusimpl.c
index b703dae..620b56d 100644
--- a/bus/ibusimpl.c
+++ b/bus/ibusimpl.c
@@ -2009,7 +2009,8 @@ bus_ibus_impl_load_global_engine_name_from_config (BusIBusImpl *ibus)
     GValue value = { 0 };
     gchar *global_engine_name = NULL;
 
-    g_assert (IBUS_IS_CONFIG (ibus->config));
+    g_assert (BUS_IS_IBUS_IMPL (ibus));
+    g_return_val_if_fail (IBUS_IS_CONFIG (ibus->config), NULL);
 
     if (ibus_config_get_value (ibus->config, "general", "global_engine", &value) &&
         G_VALUE_TYPE (&value) == G_TYPE_STRING) {
@@ -2023,7 +2024,8 @@ bus_ibus_impl_load_global_engine_name_from_config (BusIBusImpl *ibus)
 static void
 bus_ibus_impl_save_global_engine_name_to_config (BusIBusImpl *ibus)
 {
-    g_assert (IBUS_IS_CONFIG (ibus->config));
+    g_assert (BUS_IS_IBUS_IMPL (ibus));
+    g_return_if_fail (IBUS_IS_CONFIG (ibus->config));
 
     if (ibus->use_global_engine && ibus->global_engine) {
         GValue value = { 0 };
@@ -2041,7 +2043,8 @@ bus_ibus_impl_load_global_previous_engine_name_from_config (BusIBusImpl *ibus)
     GValue value = { 0 };
     gchar *global_previous_engine_name = NULL;
 
-    g_assert (IBUS_IS_CONFIG (ibus->config));
+    g_assert (BUS_IS_IBUS_IMPL (ibus));
+    g_return_val_if_fail (IBUS_IS_CONFIG (ibus->config), NULL);
 
     if (ibus_config_get_value (ibus->config, "general", "global_previous_engine", &value) &&
         G_VALUE_TYPE (&value) == G_TYPE_STRING) {
@@ -2055,7 +2058,8 @@ bus_ibus_impl_load_global_previous_engine_name_from_config (BusIBusImpl *ibus)
 static void
 bus_ibus_impl_save_global_previous_engine_name_to_config (BusIBusImpl *ibus)
 {
-    g_assert (IBUS_IS_CONFIG (ibus->config));
+    g_assert (BUS_IS_IBUS_IMPL (ibus));
+    g_return_if_fail (IBUS_IS_CONFIG (ibus->config));
 
     if (ibus->use_global_engine && ibus->global_previous_engine_name) {
         GValue value = { 0 };
-- 
1.7.1

